lib/workload/stateful/statefulStackCollectionClass.ts

```diff
@@ -0,0 +1,51 @@
+import { Construct } from 'constructs';
+import { Stack, Environment, StackProps } from 'aws-cdk-lib';
+
+import { SharedStack, SharedStackProps } from './stacks/shared/stack';
+import { TokenServiceProps, TokenServiceStack } from './stacks/token-service/deploy/stack';
+import { IcaEventPipeStack, IcaEventPipeStackProps } from './stacks/ica-event-pipe/stack';
+
+export interface StatefulStackCollectionProps {
+  sharedStackProps: SharedStackProps;
+  tokenServiceStackProps: TokenServiceProps;
+  icaEventPipeStackProps: IcaEventPipeStackProps;
+}
+
+export class StatefulStackCollection {
+  // Defined stateful stacks here
+  readonly sharedStack: Stack;
+  readonly tokenServiceStack: Stack;
+  readonly icaEventPipeStack: Stack;
+
+  constructor(
+    scope: Construct,
+    env: Environment,
+    statefulConfiguration: StatefulStackCollectionProps
+  ) {
+    this.sharedStack = new SharedStack(scope, 'SharedStack', {
+      ...this.createTemplateProps(env, 'SharedStack'),
+      ...statefulConfiguration.sharedStackProps,
+    });
+
+    this.tokenServiceStack = new TokenServiceStack(scope, 'TokenServiceStack', {
+      ...this.createTemplateProps(env, 'TokenServiceStack'),
+      ...statefulConfiguration.tokenServiceStackProps,
+    });
+
+    this.icaEventPipeStack = new IcaEventPipeStack(scope, 'IcaEventPipeStack', {
+      ...this.createTemplateProps(env, 'IcaEventPipeStack'),
+      ...statefulConfiguration.icaEventPipeStackProps,
+    });
+  }
+
+  private createTemplateProps(env: Environment, serviceName: string): StackProps {
```

Comment:
added a few more initial tagging template here for each µ-stack @victorskl @reisingerf

Reply:
Nice!